### ISSUE TITLE
12.2.3 Compiled code is not ES5

### ISSUE THREAD

**Issue by genarorg** [vercel/next.js#39090]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000
Binaries:
  Node: 14.19.1
  npm: 6.14.16
  Yarn: 1.22.19
  pnpm: N/A
Relevant packages:
  next: 12.2.3
  eslint-config-next: 12.2.3
  react: 18.2.0
  react-dom: 18.2.0

### Which example does this report relate to?

no template, just create-next-app

### Describe the Bug

We use [es-check](https://github.com/yowainwright/es-check) to make sure that our compiled code is ES5 compatible. In previous versions, we have been able to run this check on files in `.next/static/`. However, when using NextJS v12.2.3 these checks fail. 

Running the check for ES6 passes.

### Expected Behavior

I expect files compiled by NextJS using SWC to be ES5 compatible. [See this line](https://github.com/vercel/next.js/blob/84fcbd56bcb9b18e27ee57d0f7f9ba43fe0a13e9/packages/next/build/swc/options.js#L247-L248)

### To Reproduce

Commands
```shell
# Create a NextJS application using:
yarn create next-app
# Install es-check"
yarn add --dev es-check
#Run a build:
yarn build
#Check compiled files for ES5:
yarn run es-check es5 '.next/static/**/*.js'
```

Error:

```
error: ES-Check: there were 1 ES version matching errors.
info:
          ES-Check Error:
          ----
          · erroring file: .next/static/chunks/framework-f99e4e38d5204d9a.js
          · error: SyntaxError: Unexpected token (11:7018)
          · see the printed err.stack below for context
          ----

          SyntaxError: Unexpected token (11:7018)
    at Parser.pp$4.raise (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:3459:15)
    at Parser.pp$9.unexpected (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:760:10)
    at Parser.pp$9.semicolon (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:737:68)
    at Parser.pp$8.parseDoStatement (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:997:14)
    at Parser.pp$8.parseStatement (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:896:35)
    at Parser.pp$8.parseIfStatement (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:1065:28)
    at Parser.pp$8.parseStatement (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:907:35)
    at Parser.pp$8.parseBlock (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:1236:23)
    at Parser.pp$8.parseStatement (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:918:38)
    at Parser.pp$8.parseIfStatement (/Users/genarorg/work/scratch/nextjs-upgrade-2/node_modules/acorn/dist/acorn.js:1065:28)
```

**Comment by balazsorban44** [comment]
Thanks, I am investigating this. I could reproduce using `12.2.4-canary.4`, but on a local build directly from the `canary` branch, I could not. So it's likely this has been fixed already, currently trying to cut a new `canary` version to be able to verify.

**Comment by genarorg** [comment]
Thank you very much @kdy1 and @balazsorban44! can't wait to test it out in the new Canary version.

**Comment by ijjk** [comment]
Hi, the above change is now available in `v12.2.4-canary.8` of Next.js, please update and give it a try!
